projects/projects-router.js

```diff
@@ -0,0 +1,77 @@
+const express = require('express');
+
+const Projects = require('./projects-model.js');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+  try {
+    const projects = await Projects.getProjects();
+    res.json(projects);
+  } catch (err) {
+    res.status(500).json({ message: 'Failed to get Projects' });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+    const { id } = req.params;
+  
+    try {
+      const project = await Projects.findById(id);
+  
+      if (project) {
+        res.json(project);
+      } else {
+        res.status(404).json({ message: 'Could not find project with given id.' })
+      }
+    } catch (err) {
+      res.status(500).json({ message: 'Failed to get projects' });
+    }
+  });
+
+  router.post('/', async (req, res) => {
+    const projectData = req.body;
+  
+    try {
+      const project = await Projects.addProject(projectData);
+      res.status(201).json(project);
+    } catch (err) {
+      res.status(500).json({ message: 'Failed to create new project' });
+    }
+  });
+
+  router.post('/:id/actions', async (req, res) => {
```

Comment:
POST and GET actions URLs look excellent, this is exactly the syntax we want to use with the provided logic. 🔥 🔥